tests/core/ast_transforms/test_basic_sanity.py

```diff
@@ -43,6 +54,35 @@ def test_child_transform_square_to_quadruple():
     assert final == 36
 
 
+def test_child_transform_square_to_zero():
+    original = parent(2)
+    assert original == 16
+
+    result = parent(2, child=zero)
+    assert result == 0
+
+    final_two = parent(2)
+    assert final_two == 16
+
+    final = parent(3)
+    assert final == 36
+
+
+def test_child_transform_with_faked_child():
+    # maliciously faking a function's name should not affect the result
+    # this can also happen when using decorators
+    assert child_faker.__name__ == child.__name__
+
+    original = parent(1)
+    assert original == 4
+
+    result = parent(1, child=child_faker)
+    assert result == 42
```

Comment:
Fails here

Reply:
Yea, i don't think this is a valid use case - i'm happy to have this test assert an error if the `__name__` isn't the same as the function name. We are jit compiling, we have to have some restrictions :)

Reply:
As discussed, added xfail to mark as expected failure. Hopefully my next PR will fix this